mixer/test/client/env/envoy.go

```diff
@@ -70,7 +70,7 @@ func (s *TestSetup) newEnvoy() (envoy.Instance, error) {
 		options = append(options, o...)
 	}
 	/* #nosec */
-	envoyPath := filepath.Join(env.IstioBin, "envoy")
+	envoyPath := filepath.Join(env.IstioOut, "envoy")
```

Comment:
This change is simple and won't change the current, which I believe is incorrect, behavior on a Mac.

Notes on behavior for completeness: 
Today when one builds outside the container, the ${ISTIO_OUT}/envoy is a MacOS back level version of Envoy. A `make test` will of course run on the Mac and use a Mac binary (the test may/may not fail due to the back level envoy (I believe they will fail but untested by me).

If one enables `BUILD_WITH_CONTAINER=1` the process will download the Linux envoy to the darwin output directories (which I think is wrong but do have code to fix this in a comment of #19937). At that point a `make test` outside the container will fail (tries to run the linux binary on the Mac), but a `make test` in a container may work (it's running the linux envoy in the linux container`). If we actually check in the fix mentioned in the PR, I think things in the container will break since it will be pointing to ISTIO_OUT which is the Mac version. If #20338 is integrated, the test should work if we change to check the local architecture (ie. use linux if IN_BUILD_CONTAINER = 1)